fastly/kv_store.go

```diff
@@ -388,6 +395,58 @@ func (c *Client) GetKVStoreKey(i *GetKVStoreKeyInput) (string, error) {
 	return string(output), nil
 }
 
+// GetKVStoreItemInput is the input to the GetKVStoreItem function.
+type GetKVStoreItemInput struct {
+	// StoreID is the StoreID of the kv store (required).
+	StoreID string
+	// Key is the key of the item to fetch (required).
+	Key string
+}
+
+// GetKVStoreItemOutput is the output of the GetKVStoreItem function.
+type GetKVStoreItemOutput struct {
+	// Value is the value stored in the item.
+	Value string
+	// Metadata is the metadata stored in the item, if any.
+	Metadata string
+	// Generation is the generation marker of the item.
+	Generation uint64
+}
+
+// GetKVStoreItem retrieves the specified item.
+func (c *Client) GetKVStoreItem(i *GetKVStoreItemInput) (GetKVStoreItemOutput, error) {
+	if i.StoreID == "" {
+		return GetKVStoreItemOutput{}, ErrMissingStoreID
+	}
+	if i.Key == "" {
+		return GetKVStoreItemOutput{}, ErrMissingKey
+	}
+
+	path := ToSafeURL("resources", "stores", "kv", i.StoreID, "keys", i.Key)
+
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return GetKVStoreItemOutput{}, err
+	}
+	defer resp.Body.Close()
+
+	value, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return GetKVStoreItemOutput{}, err
+	}
```

Comment:
Given that KV Store values can be up to 25 MB in size, this should probably implement `io.Reader` and not `io.ReadAll()` the body into a `Value` field.

The vast majority of things stored are small strings though, so it could be reasonable to add `String() string` and `Bytes() []byte` methods on it to do it and the casting for the common cases.

Reply:
That's a good point, I hadn't thought about that when copying the existing code to make this new function. I'll need to do some investigating.